addons/pos_restaurant/models/pos_config.py

```diff
@@ -40,7 +40,8 @@ def _force_http(self):
     def get_tables_order_count(self):
         """         """
         self.ensure_one()
-        tables = self.env['restaurant.table'].search([('floor_id.pos_config_id', 'in', self.ids)])
+        floors = self.env['restaurant.floor'].search([('pos_config_id', 'in', self.ids)])
+        tables = self.env['restaurant.table'].search([('floor_id', 'in', floors.ids)])
```

Comment:
```suggestion
        tables = self.env['restaurant.table'].search([('floor_id', 'in', self.floor_ids.ids)])
```

Are archived floors still part of `self.floor_ids`?

Reply:
@caburj Thanks for the review!

I opted for two separate searches primarily for performance optimization, as this function is called frequently. Let me break down the differences:

Your approach results in multiple queries. For example:
1. `SELECT ... FROM "restaurant_floor" WHERE "restaurant_floor"."pos_config_id" IN (...)`
2. Additional `SELECT` statements fetching fields from `"restaurant_floor"`.
3. `SELECT ... FROM "restaurant_table" WHERE "restaurant_table"."floor_id" IN (...)`

This sequence leads to increased query load.

```python
floors = self.env['restaurant.floor'].search([('pos_config_id', 'in', self.ids)])
tables = self.env['restaurant.table'].search([('floor_id', 'in', floors.ids)])
```
This translates to just two efficient queries:

1. `SELECT "restaurant_floor".id FROM "restaurant_floor" WHERE (("restaurant_floor"."active" = true) AND ("restaurant_floor"."pos_config_id" IN (...)))`
2. `SELECT "restaurant_table".id FROM "restaurant_table" WHERE (("restaurant_table"."active" = true) AND ("restaurant_table"."floor_id" IN (...)))`
This method not only minimizes the number of queries but also ensures we're considering only active floors, thus maintaining accuracy and improving performance.

Moreover, using `self.floor_ids.ids` includes archived floors, contrary to our requirements. The two-search approach addresses this by filtering out archived floors effectively.